### PR TITLE
Compatible with allegro

### PR DESCRIPTION
**Pull request by presuku**
vlime compatiblity break with Allegro CL at 53355895ad02f2fba68149b1caa40dad13a447a3 and 3221f0dbd82f61cc482fa44100ca9a4c51e19c0c.
I confirmed on sbcl and Allegro CL.

### CODE REVIEW

**Comment by presuku** [comment]
~~I'm sorry. I made this pull-req by mistake.~~
At this point, this pull-req did not solve this problem.
But with the following commit, it was able to work with Allegro.

File: lisp/src/vlime-patched.lisp [hunk 1]
```diff
@@ -45,7 +45,7 @@
     (if (eql (car message) :return)
       (setf json (seq-swank-to-client json))
       (setf json (list 0 json)))
-    (let* ((yason::*char-replacements* *yason-char-replacements*)
+    (let* ((yason::*char-replacements* vlime-protocol::*yason-char-replacements*)
```
Comment (fukamachi) [inline lisp/src/vlime-patched.lisp:48]: Maybe better to export this symbol from `vlime-protocol`.

File: lisp/src/vlime-protocol.lisp [hunk 2]
```diff
@@ -88,9 +88,13 @@
...
                             :test #'eq
-                            :weakness :key
+                            #-allegro :weakness #-allegro :key
                             #+(or sbcl ecl) '(:synchronized t)
-                            #-(or sbcl ecl) '()))
+                            #-(or sbcl ecl) '(
+                              #+allegro :weak-keys t
+                              #+allegro :values t
+                              )
+                            ))
```
Comment (fukamachi) [inline lisp/src/vlime-protocol.lisp:97]: Probably other implementation supports are preferred. I'm going to add some.

**Comment by presuku** [comment]
Thanks for merging and rewriting the smarter code !

**Pull request merged**